### PR TITLE
Faster hashes

### PR DESCRIPTION
**Pull request by elcasteel**
# Description
Use the faster hashes that we use when snapshotting most resources for routes as well.

# Context

This is one of the first quick wins we found as part of our scale testing.

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5621fa

### CODE REVIEW

File: projects/gloo/pkg/syncer/sanitizer/route_replacing_sanitizer_test.go [hunk 1]
```diff
@@ -250,7 +251,7 @@ var _ = Describe("RouteReplacingSanitizer", func() {
 		listenersWithFallback := listeners.Items[fallbackListenerName]
 		clustersWithFallback := clusters.Items[fallbackClusterName]
 
-		Expect(sanitizedRoutes.ResourceProto()).To(Equal(expectedRoutes))
+		Expect(sanitizedRoutes.ResourceProto()).To(matchers.MatchProto(expectedRoutes))
```
Comment (elcasteel) [inline projects/gloo/pkg/syncer/sanitizer/route_replacing_sanitizer_test.go:254]: There's a similar change to the tests in this PR: https://github.com/solo-io/gloo/pull/4088/files so I think this is still testing for the correct behavior

File: projects/gloo/pkg/translator/translator_test.go [hunk 2]
```diff
@@ -1773,11 +1774,12 @@ var _ = Describe("Translator", func() {
...
+
 			Expect(metadata.Fields).To(HaveLen(4))
-			Expect(metadata.Fields[dc(east)]).To(Equal(trueValue))
-			Expect(metadata.Fields[dc(west)]).To(Equal(falseValue))
-			Expect(metadata.Fields[tag(dev)]).To(Equal(falseValue))
-			Expect(metadata.Fields[tag(prod)]).To(Equal(trueValue))
+			Expect(metadata.Fields[dc(east)].GetKind()).To(Equal(trueValue.GetKind()))
+			Expect(metadata.Fields[dc(west)].GetKind()).To(Equal(falseValue.GetKind()))
+			Expect(metadata.Fields[tag(dev)].GetKind()).To(Equal(falseValue.GetKind()))
+			Expect(metadata.Fields[tag(prod)].GetKind()).To(Equal(trueValue.GetKind()))
```
Comment (EItanya) [inline projects/gloo/pkg/translator/translator_test.go:1782]: You can use `MatchProto` here as well I think

**Review by bewebi — APPROVED** [review]
LGTM

**Review by bewebi — APPROVED** [review]
LGTM

**Pull request merged**